cloudant-sync-datastore-core/src/main/java/com/cloudant/sync/datastore/QueryableDatastore.java

```diff
@@ -0,0 +1,282 @@
+        this(dir, name, new NullKeyProvider());
+    }
+
+    public QueryableDatastore(String dir, String name, KeyProvider provider) throws SQLException, IOException, DatastoreException {
+        super(dir, name, provider);
+
+        validFieldName = Pattern.compile(INDEX_FIELD_NAME_PATTERN);
+
+        final String filename = this.extensionDataFolder(EXTENSION_NAME) + File.separator
+                + "indexes.sqlite";
+
+        dbQueue = new SQLDatabaseQueue(filename, provider);
+        dbQueue.updateSchema(new SchemaOnlyMigration(QueryConstants.getSchemaVersion1()), 1);
+        dbQueue.updateSchema(new SchemaOnlyMigration(QueryConstants.getSchemaVersion2()), 2);
+        textSearchEnabled = ftsAvailable(dbQueue);
+
+    }
+
+    /**
+     * Check that support for text search exists in SQLite by building a VIRTUAL table.
+     *
+     * @return text search enabled setting
+     */
+    public static boolean ftsAvailable(SQLDatabaseQueue q) {
+        boolean ftsAvailable = false;
+        Future<Boolean> result = q.submitTransaction(new SQLQueueCallable<Boolean>() {
+            @Override
+            public Boolean call(SQLDatabase db) throws Exception {
+                List<String> statements = new ArrayList<String>();
+                statements.add(String.format("CREATE VIRTUAL TABLE %s USING FTS4 ( col )",
+                        FTS_CHECK_TABLE_NAME));
+                statements.add(String.format("DROP TABLE %s", FTS_CHECK_TABLE_NAME));
+
+                for (String statement : statements) {
+                    db.execSQL(statement);
+                }
+                // If we made it this far, FTS is enabled, otherwise `execSQL` will throw.
+                // Logging happens in calling method.
+                return  Boolean.TRUE;
+            }
+        });
+
+        try {
+            ftsAvailable = result.get();
+        } catch (ExecutionException e) {
+            logger.log(Level.SEVERE, "Execution error encountered:", e);
+        } catch (InterruptedException e) {
+            logger.log(Level.SEVERE, "Execution interrupted error encountered:", e);
+        }
+
+        return ftsAvailable;
+    }
+
+    @Override
+    public Map<String, Object> listIndexes() {
+        try {
+            return dbQueue.submit(new SQLQueueCallable<Map<String, Object> >() {
+                @Override
+                public Map<String, Object> call(SQLDatabase database) throws Exception {
+                    return QueryableDatastore.listIndexesInDatabase(database);
+                }
+            }).get();
+        } catch (InterruptedException e) {
+            logger.log(Level.SEVERE,"Failed to list indexes",e);
+            throw new RuntimeException(e);
+        } catch (ExecutionException e) {
+            logger.log(Level.SEVERE,"Failed to list indexes",e);
+            throw new RuntimeException(e);
+        }
+
+    }
+
+    public static Map<String, Object> listIndexesInDatabase(SQLDatabase db) {
```

Comment:
I know this is mostly a copy/paste, but I don't see the point of this static method I think a `ListIndexesCallable` would be better. Although I guess that probably falls into the remit of #291.